mlflow/models/resources.py

```diff
@@ -0,0 +1,145 @@
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from dataclasses import dataclass
+from enum import Enum
+from typing import List
+
+
+class ResourceType(Enum):
+    DATABRICKS_VECTOR_SEARCH_INDEX_NAME = "databricks_vector_search_index_name"
```

Comment:
The `_NAME` suffix for a `ResourceType` is a bit confusing:

```suggestion
    DATABRICKS_VECTOR_SEARCH = "databricks_vector_search_index"
```

Can we make this change for the other instances too?